toucan_connectors/google_big_query/google_big_query_connector.py

```diff
@@ -37,8 +43,21 @@ def _define_type(value) -> str:
         return 'FLOAT64'
     elif isinstance(value, str):
         return 'STRING'
+    # TODO - check bad return type
+    return 'STRING'
+
+
+def _define_array_type(name: str, values: List[Any]) -> BigQueryParam:
+    return bigquery.ArrayQueryParameter(
+        name, _define_scalar_type(values[0] if len(values) > 0 else ''), values
```

Comment:
Obviously this will only work for arrays that have all elements of the same type.
It should probably be the case for the vast majority though, so it seems OK.

Reply:
Yeah that's flaky. Worst thing is that it probably won't fail if we have an array of mixed types, since everything gets serialized as JSON again. So everything will probably be cast to a string or something like that.

And the bigquery lib seems to do the same as us here :stuck_out_tongue: https://github.com/googleapis/python-bigquery/blob/35aeaa61db271027cb95d68de362544e437dd761/google/cloud/bigquery/dbapi/_helpers.py#L119